pallets/roles/src/impls.rs

```diff
@@ -294,12 +333,150 @@ impl<T: Config> Pallet<T> {
 		<Ledger<T>>::insert(staker, ledger);
 	}
 
+	/// This function retrieves the current era and the total rewards available for distribution
+	/// from the runtime configuration. It then computes rewards for active validators and
+	/// validators based on their role types. After computing rewards, it adds individual reward
+	/// points to each eligible validator and records them in the staking pallet ErasRewardPoints
```

Comment:
As mentioned in slack I think that modifying the staking pallet's rewards opens up an attack vector against the entire system's incentives. Currently validators get staking rewards based on points. If a malicious restaking validator can abuse how many points they receive by artificially creating jobs for themselves then they can hijack the base staking rewards as well, no?

Reply:
I recall saying this was why we need to duplicate the points system here and maintain this duplicate exclusively for restaking incentive inflation. This would at least shield cross-incentive system attacks (manipulating staking rewards based on how restaking rewards are calculated). I intended for them to be separate in the design.